### PR TITLE
Add warning suppression(rawtypes) to generated CustomType#javaType method

### PR DESCRIPTION
**Pull request by xrd**
When using javac with warnings as errors, CustomTypes fail compilation with rawtypes error. This adds  a switch to the apollo plugin which can add the proper annotation (`@SuppressWarning("rawtypes")`) to the generated custom type file.

### CODE REVIEW

**Comment by xrd** [comment]
Summary is at issue #926, and sample code demonstrating the error is here: https://github.com/xrd/apollo_custom_types_error

**Comment by sav007** [comment]
Thanks for the PR. There is one failing test `JavaTypeResolverTest `. Plus I think that you should add tests for compiler, look at https://github.com/apollographql/apollo-android/tree/master/apollo-compiler/src/test/graphql/com/example

To create a new test for compiler you need to:
0. Install the code gen version `npm install -g apollo-codegen@0.19.1`
1.  Create a new folder under `/apollo-compiler/src/test/graphql/com/example`
2. Create `TestOperation.graphql` file with the query
3. Run `./gradlew :apollo-compiler:updateTestIRFiles`
4. Run `com.apollographql.apollo.compiler.CodeGenTest` unit tests
5. Put generate java files from `/apollo-compiler/build/generated/source/apollo/com/example/your-folder` to `/apollo-compiler/src/test/graphql/com/example/your-folder`
6. Run again `com.apollographql.apollo.compiler.CodeGenTest` to verify all tests are green

**Comment by xrd** [comment]
@sav007 Great, thanks for the help. I believe I have fixed the failing test and added another which verifies my compiler output is correct when the `useRawTypesWarningSuppression` apollo plugin switch is provided.

File: README.md [hunk 1]
```diff
@@ -211,14 +211,14 @@ ApolloClient.builder()
...
-turned on, your custom type will not compile. You can add a switch `useRawTypesSuppression` to the
+turned on, your custom type will not compile. You can add a switch `useRawTypesWarningSuppression` to the
 apollo plugin configuration which will annotate your generated class with the proper suppression
 (`@SuppressWarnings("rawtypes")`:
 
 ```
 apollo {
     customTypeMapping['URL'] = "java.lang.String"
-    useRawTypesSuppression = "true"
+    useRawTypesWarningSuppression = "true"
```
Comment (digitalbuddha) [inline README.md]: Nitpick: `supressRawTypesWarning` sounds better imo

Comment (xrd) [reply README.md]: 👍

**Comment by xrd** [comment]
@digitalbuddha and @sav007 do I need to do anything else? Not in a rush to get this merged, but let me know if there is anything needed on my end.

File: apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/Annotations.kt [hunk 2]
```diff
@@ -12,5 +12,7 @@ object Annotations {
   val OVERRIDE: AnnotationSpec = AnnotationSpec.builder(Override::class.java).build()
   val GENERATED_BY_APOLLO: AnnotationSpec = AnnotationSpec.builder(Generated::class.java)
       .addMember("value", CodeBlock.of("\$S", "Apollo GraphQL")).build()
+  val SUPPRESS_WARNINGS: AnnotationSpec = AnnotationSpec.builder(SuppressWarnings::class.java)
+          .addMember( "value", CodeBlock.of( "\$S", "rawtypes")).build()
```
Comment (hwillson) [inline apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/Annotations.kt]: Can you adjust the indentation here to line up with the rest of the file?

File: apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/CustomEnumTypeSpecBuilder.kt [hunk 3]
```diff
@@ -37,10 +37,15 @@ class CustomEnumTypeSpecBuilder(
...
-              .addStatement("return \$T.class", javaTypeName.toJavaType())
+                  .addModifiers(Modifier.PUBLIC)
+                  .apply {
+                    if (context.suppressRawTypesWarning) {
+                      this.addAnnotation(com.apollographql.apollo.compiler.Annotations.SUPPRESS_WARNINGS)
+                    }
+                  }
+                .addAnnotation(Override::class.java)
+                .returns(Class::class.java)
+                .addStatement("return \$T.class", javaTypeName.toJavaType())
```
Comment (hwillson) [inline apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/CustomEnumTypeSpecBuilder.kt]: The indentation seems to be off here as well - can you adjust it to match the rest of the file?

File: apollo-compiler/src/test/graphql/com/example/custom_scalar_type_warnings/custom_scalar_type_warnings/TestQuery.java [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Generated;
+import org.jetbrains.annotations.Nullable;
+
+@Generated("Apollo GraphQL")
+public final class TestQuery implements Query<TestQuery.Data, Optional<TestQuery.Data>, Operation.Variables> {
+  public static final String OPERATION_DEFINITION = "query TestQuery {\n"
+      + "  links\n"
+      + "}";
```
Comment (hwillson) [inline apollo-compiler/src/test/graphql/com/example/custom_scalar_type_warnings/custom_scalar_type_warnings/TestQuery.java]: Please keep indentation to 2 space here.

**Comment by xrd** [comment]
@hwillson I've pushed the changes to fix the formatting. Thanks!

**Pull request merged**